index.js

```diff
@@ -1,21 +1,35 @@
-// Inspired by (but complete rewrite of) requirejs-metagen
-let _ = require('lodash/fp')
-let Promise = require('bluebird')
-let readDir = Promise.promisify(require('recursive-readdir'))
-let fs = Promise.promisifyAll(require('fs'))
-
-// Path Utils
-let filesRelative = (dir, ex) => readDir(dir, ex).map(x => x.slice(dir.length))
+// Inspired by (but complete rewrite of) requirejs-metagen.
+let _       = require('lodash/fp') // Add lodash helper functions.
+let Promise = require('bluebird')  // Replace promises with the Bluebird Promises library. ("Bluebird cuts corners" -- yikes!)
+let readDir = Promise.promisify(require('recursive-readdir')) // Lists all files in a certain directory and all its sub-directories.
+let fs      = Promise.promisifyAll(require('fs')) // Add the filesystem library.
+
+// Define path Utils.
+
+// Get all the file names in a certain directory and its sub-directories, excluding the directory path.
+let filesRelative = (dir, exclusions) => readDir(dir, exclusions).map(filename => filename.slice(dir.length))
```

Comment:
Instead of being more explicit with a variable name like `filename`, we actually should aim for point free syntax and pull out a function like `let slice = start => thing => thing.slice(start)` and possibly pull out `futil-js`. I'm not sure why we didn't do that in the first place actually.